epoch/api/naming_system_api_usage.md

```diff
@@ -0,0 +1,95 @@
+# Naming System - intended usage
+
+## Introduction
+
+The way to interact with Naming System is to use HTTP API.
+To read about possible states and life cycle of a name please see [the doc](/AENS.md).
+
+The following assumes that the node exposes at address 127.0.0.1 the following ports:
+* HTTP external API: 3013
+* HTTP internal API: 3113
+
+## Name management flow
+
+Check the public key of the node first (used as account key in later API calls):
+```
+curl http://localhost:3113/v2/account/pub-key
+{"pub_key":"ak$3scLu3oJbhsdCJkDjfJ6BUPJ4M9ZZJe57CQ56deSbEXhaTSfG3Wf3i2GYZV6APX7RDDVk4Weewb7oLePte3H3QdBw4rMZw"}
+```
+
+Verify that the name that your going to claim is not taken.
+For the purpose of this example let's consider `fooo.baar.aet` name:
+```
+curl http://localhost:3013/v2/name\?name\=fooo.baar.aet
+{"reason":"Name not found"}
+```
+
+In order to claim a name you need to submit a preclaim transaction first, containing commitment hash.
+
+You need to pick a random salt value, to make commitment hash unique.
+Commitment hash is calculated based on [the formula](/AENS.md#pre-claim).
+For the purpose of this example name `fooo.baar.aet` with salt 123 is considered:
+```
+Base58(Commitment(fooo.baar.aet, 123)) = "cm$2fxgad7CSY4dPsn6SVtgdigNq8gdzRJb9W3VB1W5oRKHqC3yH6"
+```
+
+To preclaim a name, send name preclaim transaction with commitment hash in the payload:
+```
+curl -X POST -H "Content-Type: application/json" http://localhost:3113/v2/name-preclaim-tx -d '{"commitment": "cm$2fxgad7CSY4dPsn6SVtgdigNq8gdzRJb9W3VB1W5oRKHqC3yH6", "fee": 1}'
+{"commitment":"cm$2fxgad7CSY4dPsn6SVtgdigNq8gdzRJb9W3VB1W5oRKHqC3yH6"}
+```
+
+Mind that preclaim and claim transactions will not be accepted in the same block,
+so please check that at least one block is mined before submitting claim transaction.
+To check the height of the blockchain use the following call:
+```
+curl http://localhost:3013/v2/top
+{"hash":"bh$8CBh7DAfnsbgZf8Btd541jm1Smhd6177CBv7UgK8vcAgxfH8m","height":17,"nonce":10943666989495949657,"pow":[1042489,1586082,7185269,17390250,22010949,23823133,26084833,26496880,26982374,27985078,29155969,30765562,34280527,36576404,40341757,42010247,48864118,53081230,53085626,58374655,58640182,60230517,63786551,65401252,78883695,80142167,85509843,87676973,91934825,92005073,94576137,100347674,103149860,106024683,109781247,113905034,121009561,124198978,126781361,129629191,131266393,131659086],"prev_hash":"bh$2fDH24kVtC4UuEheXMeSZhqMUoaDyo495iotrCVYv1ES8wH38K","state_hash":"bs$2jVx7kz9Uvp55etQsoJTmKwy2pMAHC2tn5dUt5W2gq99k7UKWp","target":539923902,"time":1517504077312,"txs_hash":"bx$2RLzQbRE1frHPYnPi8peZgBzeUGccV4NtWhDf6tGnn6p3wMSZv","version":5}
+```
+
+When a name is preclaimed, you are in a position to claim it.
+You must use the name with the same salt as used in commitment hash computation:
+```
+curl -X POST -H "Content-Type: application/json" http://localhost:3113/v2/name-claim-tx -d '{"name": "fooo.baar.aet", "name_salt": 123, "fee": 1}'
```

Comment:
I am testing this on two nodes.
I tested posting the pre-claim tx on a node, then attempting to post the claim tx on another node (hence account origin of claim distinct from account origin of pre-claim). The API returned 200, tx is still stuck in mempool as semantically invalid (that is correct and expected).
Potential user experience improvement may be preventing posting a claim tx unless the corresponding (???) pre-claim tx in the chain or in mempool? Though it may have corner cases so best to keep this way for this release anyway.

Then I tested posting claim on same node as pre-claim and it removed *this* tx from mempool - that is correct and expected. All good so far.